sqlmesh/core/analytics/__init__.py

```diff
@@ -9,6 +9,12 @@
 from sqlmesh.core.analytics.dispatcher import AsyncEventDispatcher, NoopEventDispatcher
 from sqlmesh.utils import str_to_bool
 
+if t.TYPE_CHECKING:
+    from typing_extensions import ParamSpec
```

Comment:
I think we need something like
```
if sys.version_info >= (3, 9):
    from typing import ParamSpec
else:
    from typing_extensions import ParamSpec
```

Reply:
What's the rationale for doing that? The `typing_extensions` module already has logic to determine whether it should just import the corresponding class in the `typing` module vs. recreating the type itself. It seems best to let it handle that instead of trying to redo it.

Reply:
@izeigerman I pushed a change similar to what you suggested. Hopefully that unblocks this.